src/input.cpp

```diff
@@ -1262,28 +1261,21 @@ cata::optional<tripoint> input_context::get_coordinates( const catacurses::windo
     if( !coordinate_input_received ) {
         return cata::nullopt;
     }
-    int view_columns = getmaxx( capture_win );
-    int view_rows = getmaxy( capture_win );
-    int win_left = getbegx( capture_win ) - VIEW_OFFSET_X;
-    int win_right = win_left + view_columns - 1;
-    int win_top = getbegy( capture_win ) - VIEW_OFFSET_Y;
-    int win_bottom = win_top + view_rows - 1;
-    if( coordinate_x < win_left || coordinate_x > win_right || coordinate_y < win_top ||
-        coordinate_y > win_bottom ) {
+    const point view_size( getmaxx( capture_win ), getmaxy( capture_win ) );
+    const point win_min( getbegx( capture_win ) - VIEW_OFFSET_X,
+                         getbegy( capture_win ) - VIEW_OFFSET_Y );
+    const rectangle win_bounds( win_min, win_min + view_size );
+    if( !win_bounds.contains_half_open( coordinate ) ) {
         return cata::nullopt;
     }
 
-    int view_offset_x = 0;
-    int view_offset_y = 0;
+    point view_offset = point_zero;
```

Comment:
Explicit initialization is not needed here. The other equivalent code in "src/sdltiles.cpp" does not have the explicit initialization either.
```suggestion
    point view_offset;
```